crates/starknet-devnet-core/src/state/mod.rs

```diff
@@ -466,23 +470,22 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
```

Comment:
Something needs to be done about this.

Reply:
Example of working test:
```rust
#[test]
    fn correct_state_diff() {
        let mut state = setup();
        let addr = starknet_api::core::ContractAddress::from(1u8);
        state.get_class_hash_at(addr);
        state.set_class_hash_at(addr, starknet_api::core::ClassHash(Felt::TWO)).unwrap();

        let a = state.state.to_state_diff().unwrap();
        println!("{:?}", a);
    }
```

Reply:
state.get_class_hash_at writes to the cache initial reads. Computing StateDiff uses strict_subtract_mapping with cache.writes and cache.initial_reads as its arguments. Which is a required step for the strict_subtract_mapping method to return without an error. Getting the different values for each key and each key have to be part of the collection that is being subtracted

Reply:
Nice finding. Is your suggestion to add these reads to all ignored tests?

Reply:
probably, but didn't investigated why the integration tests doesn't fail.

Reply:
We are probably somewhere implicitly reading the state, likely somewhere in the historic state construction, haven't yet checked this idea.

Reply:
Added the reads before writes: https://github.com/0xSpaceShard/starknet-devnet-rs/pull/557/commits/e195917cdc8bbe64a022bebee20f2d7b90a7d751

Also used these reads to assert that the state is indeed untouched before we write.